src/test/java/jenkins/plugins/git/GitSCMSourceTraitsTest.java

```diff
@@ -137,8 +137,18 @@ public void pimpped_out() throws Exception {
                                         hasProperty("localBranch", is("**"))
                                 )
                         ),
-                        Matchers.<SCMSourceTrait>instanceOf(CleanAfterCheckoutTrait.class),
```

Comment:
Can you have a look if there are tests that cover reading old job configurations and maybe adapt/add such?
I am asking because it would be good to have tests that make sure that the "old" and "new" configuration XML (added property/setter) can be read properly (backward compatibility) for traits and extensions.

old
```
<extensions>
   <hudson.plugins.git.extensions.impl.CleanBeforeCheckout/>
</extensions>
```

new
```
<extensions>
   <hudson.plugins.git.extensions.impl.CleanBeforeCheckout>
      <deleteUntrackedNestedRepositories>true</deleteUntrackedNestedRepositories>
   </hudson.plugins.git.extensions.impl.CleanBeforeCheckout>
</extensions>
```

Reply:
[This](https://github.com/jenkinsci/git-plugin/pull/792/commits/337e22ecbd77b228572cba9f51301873741a405a) is OK?

Reply:
This already looks ok. May I ask to you to additionally reduce the XML to the required minimum? So it's more obvious what this test is really about and it doesn't fail in cases that are not relevant for the backward compatibility of these two traits.

Reply:
Reduced to:

```
<jenkins.plugins.git.GitSCMSource>
  <extensions>
    <hudson.plugins.git.extensions.impl.CleanCheckout />
    <hudson.plugins.git.extensions.impl.CleanBeforeCheckout />
  </extensions>
</jenkins.plugins.git.GitSCMSource>
```